Xamarin.Forms.Platform.WPF/WPFPlatformServices.cs

```diff
@@ -151,6 +151,18 @@ public void StartTimer(TimeSpan interval, Func<bool> callback)
 			};
 		}
 
+		public void StartTimer(TimeSpan interval, Func<Task<bool>> callback)
+		{
+			var timer = new DispatcherTimer(DispatcherPriority.Background, System.Windows.Application.Current.Dispatcher) { Interval = interval };
```

Comment:
Is this pushes the callback to the background thread? The other implementations attempt to run the callback function on the main thread. I see that this mimics the behavior of the non-async API. Not sure if this is something that should be consistent across the platforms or if there is a technical limitation

Reply:
I'm not a WPF expert, but I it looks like this runs on the main thread.

I first thought it was running on a background thread too because of `DispatcherPriority.Background`. However the docs indicate that [`Dispatcher.Background`](https://docs.microsoft.com/dotnet/api/system.windows.threading.dispatcherpriority?view=netframework-4.8) is its priority, meaning specifically that this task will execute on the passed in thread (`Application.Current.Dispatcher`, i.e. the [Main Thread](https://stackoverflow.com/a/11625264/5953643)) at this priority:
> The enumeration value is 4. Operations are processed after all other non-idle operations are completed.